libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/VariableUsage.kt

```diff
@@ -1,13 +1,34 @@
 package com.apollographql.apollo3.ast
 
+import com.apollographql.apollo3.annotations.ApolloDeprecatedSince
+import com.apollographql.apollo3.annotations.ApolloDeprecatedSince.Version.v4_0_0
+
 /**
  * A variable used in a [GQLValue]
  */
 class VariableUsage(
     val variable: GQLVariableValue,
     val locationType: GQLType,
-    val hasLocationDefaultValue: Boolean
-)
+    val hasLocationDefaultValue: Boolean,
+    val isOneOfInputObject: Boolean,
+) {
+  @Deprecated(
+      message = "Use the constructor with isOneOfInputObject instead",
+      replaceWith = ReplaceWith("VariableUsage(variable, locationType, hasLocationDefaultValue, isOneOfInputObject = false)"),
+      level = DeprecationLevel.ERROR
+  )
+  @ApolloDeprecatedSince(v4_0_0)
+  constructor(
```

Comment:
apollo-ast has many API breakages in 4.0. This one feels like a small one, I would just go for the new constructor directly

Reply:
Fair enough!